Indicators/AverageDirectionalIndex.cs

```diff
@@ -0,0 +1,332 @@
+                    TrueRange.Reset();
+                }
+                );
+
+            SmoothedTrueRange = new FunctionalIndicator<IndicatorDataPoint>(name + "_SmoothedTrueRange",
+                    currentBar => ComputeSmoothedTrueRange(period),
+                    isReady => _previousInput != null
+                );
+
+
+            SmoothedDirectionalMovementPlus = new FunctionalIndicator<IndicatorDataPoint>(name + "_SmoothedDirectionalMovementPlus",
+                    currentBar => ComputeSmoothedDirectionalMovementPlus(period),
+                    isReady => _previousInput != null
+                );
+
+            SmoothedDirectionalMovementMinus = new FunctionalIndicator<IndicatorDataPoint>(name + "_SmoothedDirectionalMovementMinus",
+                    currentBar => ComputeSmoothedDirectionalMovementMinus(period),
+                    isReady => _previousInput != null
+                );
+        }
+
+        /// <summary>
+        /// Computes the Smoothed Directional Movement Plus value.
+        /// </summary>
+        /// <param name="period">The period.</param>
+        /// <returns></returns>
+        private decimal ComputeSmoothedDirectionalMovementPlus(int period)
+        {
+
+            decimal value;
+
+            if (Samples < period)
+            {
+                value = SmoothedDirectionalMovementPlus.Current + DirectionalMovementPlus.Current;
+            }
+            else
+            {
+                value = SmoothedDirectionalMovementPlus.Current - (SmoothedDirectionalMovementPlus.Current / period) + DirectionalMovementPlus.Current;
+            }
+
+            return value;
+        }
+
+        /// <summary>
+        /// Computes the Smoothed Directional Movement Minus value.
+        /// </summary>
+        /// <param name="period">The period.</param>
+        /// <returns></returns>
+        private decimal ComputeSmoothedDirectionalMovementMinus(int period)
+        {
+            decimal value;
+
+            if (Samples < period)
+            {
+                value = SmoothedDirectionalMovementMinus.Current + DirectionalMovementMinus.Current;
+            }
+            else
+            {
+                value = SmoothedDirectionalMovementMinus.Current - (SmoothedDirectionalMovementMinus.Current / 14) + DirectionalMovementMinus.Current;
+            }
+
+            return value;
+        }
+
+        /// <summary>
+        /// Computes the Smoothed True Range value.
+        /// </summary>
+        /// <param name="period">The period.</param>
+        /// <returns></returns>
+        private decimal ComputeSmoothedTrueRange(int period)
+        {
+            decimal value;
+
+            if (Samples < period)
+            {
+                value = SmoothedTrueRange.Current + TrueRange.Current;
+            }
+            else
+            {
+                value = SmoothedTrueRange.Current - (SmoothedTrueRange.Current / period) + TrueRange.Current;
```

Comment:
Maybe our current implementation of Wilder's smoothing is incorrect. I think this is what he used to do, SMA until the period is full and then the kind of exponential corrector for each new point